src/test/java/net/snowflake/client/jdbc/SnowflakeSerializableTest.java

```diff
@@ -0,0 +1,381 @@
+package net.snowflake.client.jdbc;
+
+import static net.snowflake.client.jdbc.SnowflakeChunkDownloader.NoOpChunkDownloader;
+import static net.snowflake.client.jdbc.SnowflakeResultSetSerializableV1.ChunkFileMetadata;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import net.snowflake.client.core.ObjectMapperFactory;
+import net.snowflake.client.core.QueryResultFormat;
+import net.snowflake.client.core.SFBaseSession;
+import net.snowflake.client.core.SFBaseStatement;
+import net.snowflake.client.core.SFStatementType;
+import org.junit.Test;
+
+public class SnowflakeSerializableTest {
```

Comment:
We already have integration tests for `SnowflakeResultSetSerializableV1` for sake of regression. As for this change it can really only be tested by unit tests since this new `SnowflakeRichResultSetSerializableV1` is not really tied to any result set fetching.